drivers/properties/sensor/sensor-device.yaml

```diff
@@ -0,0 +1,10 @@
+properties:
+  vendor:
+    type: string
+    required: true
+
+  model:
+    type: string
+    required: true
```

Comment:
both properties can be obtained from DT. From dt spec `compatible`:

> The recommended format is "manufacturer,model", where manufacturer is a string describing
the name of the manufacturer (such as a stock ticker symbol), and model specifies the model number.

Human readable string for manufacturers is found in `dts/bindings/vendor-prefixes.txt`. Summarizing, these properties already exist. There's actually ongoing work to expose them https://github.com/zephyrproject-rtos/zephyr/pull/50015

Reply:
I used these properties for the example based on this PR https://github.com/zephyrproject-rtos/zephyr/pull/49294 which intends to add these values.

The edtlib can be expanded further to also use the `dts/bindings/vendor-prefixes.txt` and compatible to look up the full name and generate a macro for it, but is that really a job for the edtlib?

The properties files should be used for immutable base properties for devices, the number of channels a PWM controller contains, the number of alarms an RTC supports, the specific data a sensor can return etc. These are base properties that are immutable, like vendor and model, which perfectly fit into the purpose of the properties files. With vendor and model however, it just happens it can be extracted from the compatible property.

I think both methods of getting vendor and model are fine, I prefer moving them to the properties files because they are designed for it, it requires more processing to get it from the edtlib.

Reply:
> The properties files should be used for immutable base properties for devices, the number of channels a PWM controller contains, the number of alarms an RTC supports, the specific data a sensor can return etc. These are base properties that are immutable, like vendor and model, which perfectly fit into the purpose of the properties files. 

Beware that the properties you describe above may not be "immutable" or rather constant across the devices supported by a given driver. Examples off the top of my head:
- The NXP MCUX FlexCAN controller driver supports the FlexCAN IP core on multiple different SoCs, but the number of available mailboxes vary from SoC to SoC.
- The TI LMP90xxx ADC driver supports an entire family of ADCs, but the resolution, number of channels etc. vary across the family.
- Same goes the the Microchip MCP320x ADC driver.

I am sure the above is true for many other in-tree drivers, e.g. many sensor drivers support multiple variants of a given sensor, each with different properties.

Reply:
> The properties files should be used for immutable base properties for devices, the number of channels a PWM controller contains, the number of alarms an RTC supports, the specific data a sensor can return etc.

These example properties, I might add, sound more like a hardware description than a driver description and properly belong in the devicetree.

Reply:
Sure, properties files are not a good solution for hardware model revision or layout specific properties. The are useful for information regarding a device driver and its capabilities, like which specific API it supports, which features it supports, which hardware revisions it supports etc.